test/bases.py

```diff
@@ -9,15 +9,30 @@
 from worlds import AutoWorld
 from worlds.AutoWorld import World, call_all
 
-from BaseClasses import Location, MultiWorld, CollectionState, ItemClassification, Item
+from BaseClasses import CollectionState, Item, ItemClassification, Location, MultiWorld, PathValue, Region
 from worlds.alttp.Items import item_factory
 
 
+class DefaultOptionsDict(typing.Dict[str, typing.Any]):
+    def __setitem__(self, key: str, value: typing.Any) -> None:
+        raise Exception("Can not modify global default options. Please copy in setUp.")
+
+    def __copy__(self) -> typing.Dict[str, typing.Any]:
```

Comment:
Is it not weird that copying an object return you a different type?

Reply:
See above answer. Dynamically creating the options in setUp is legit imo. I agree that checksmate may be better done by providing static options instead of copy, and i am open to reflect that in the Exception message, but that does not mean there isn't a case where wouldn't make sense for a subclass of world test to do finicky stuff.

The use of a custom class (rather than frozendict) is to be able to provide a better error message.

> Is it not weird that copying an object return you a different type?

It is, but mypy doesn't complain[^1], and I think the use of `copy(other)` is a better pattern than `{k: v for k, v in other.items()}` and `dict(other.items))` - i.e. the custom dict is to be used like a regular dict but adds a safety net.

[^1]: for x: BaseType = DerivedType(), copy(x) should expect BaseType

Reply:
I guess it does make sense to create a custom class then. The error message would definitely be helpful to devs and hopefully guide them to the right intended usage of the options.  

Sure, `copy(other)` is a better pattern than the other examples you found, but `dict(other)` or `{**other}` are just as good IMO. I would even argue that `dict(other)` is better, since it looks like a type convention, which it is because `other` is not exactly a `dict`, because it's immutable. Since `__copy__` would not need to be implemented, users would not be guided toward creating copies of the options, which again steer them in the right direction.